android-json-form-wizard/src/main/java/com/vijay/jsonwizard/fragments/JsonFormFragment.java

```diff
@@ -380,10 +376,18 @@ public void hideKeyBoard() {
 
     @Override
     public void transactThis(JsonFormFragment next) {
+        if (getFragmentManager().getBackStackEntryCount() > 0) { //Note: first step is not usually added to backStackEntry
+            FragmentManager.BackStackEntry stackEntry = getFragmentManager().getBackStackEntryAt(getFragmentManager().getBackStackEntryCount() - 1);
+            if (StringUtils.isNotBlank(stackEntry.getName()) &&
+                    stackEntry.getName().equalsIgnoreCase(next.getArguments().getString(JsonFormConstants.STEPNAME))) {
+                return;
+            }
```

Comment:
What is this doing? Please add a comment.

Reply:
It prevents a very rare edge case whereby same fragment gets added twice in the fragment stack

Reply:
Please add this as a comment, it wouldn't be obvious unless you explained it to someone.

Reply:
ok